### PR TITLE
Move the digest email opt-in to the follow tags onboarding step

### LINKED ISSUE #19986: Move the digest email opt-in to the follow tags onboarding step
## Is this feature related to a problem? Please describe.

- As a new user, I want to subscribe to a digest of the top posts for the tags I follow so I can be kept up to date on content related to my interests, even when I'm not actively checking the community.
- The current opt-in screen for digest emails is separated from the context in which users select which tags they are interested in. To encourage higher sign-up rates for these emails, we would like to move the opt-in to the tags screen.

## Describe the solution you’d like

- Move the digest opt-in to the bottom of the existing tags onboarding step, following the designs (but not yet implementing the main tag redesign)
- The opt-in should appear over the scrollable list of tags so it is always visible.
- The opt-in should use the following wording and will not be configurable:
    > **Get a Periodic Digest of Top Posts** 
    > We'll email you with a curated selection of top posts based on the tags you follow.

## Designs

 - [Desktop Designs](https://www.figma.com/file/u7VcNG4pjbn7K01V8JrFWo/Member-Onboarding-Process?type=design&node-id=2951-6338&mode=design&t=r6NBtzaywCXYCyL6-4)
 - [Mobile Designs](https://www.figma.com/file/u7VcNG4pjbn7K01V8JrFWo/Member-Onboarding-Process?type=design&node-id=2951-6545&mode=design&t=Zaqhvd8jQUE0bVep-4)

## Definition of Done

- [ ] Move the digest opt-in to the bottom of the tags onboarding step, ensuring it remains visible over any tags displayed on the step
- [ ] Clicking anywhere on opt-in should activate the checkbox.
- [ ] Update the styling to match the designs

## Additional Context

- We have moved the navigation to the bottom of the modal to help reader flow and comprehension and to ensure users get an opportunity to read all of the content in the onboarding flow.

### PR DESCRIPTION
**Pull request by rt4914**
## What type of PR is this? (check all applicable)

- [x] Refactor

## Description

[Desktop Designs](https://www.figma.com/file/u7VcNG4pjbn7K01V8JrFWo/Member-Onboarding-Process?type=design&node-id=2951-6338&mode=design&t=r6NBtzaywCXYCyL6-4)
[Mobile Designs](https://www.figma.com/file/u7VcNG4pjbn7K01V8JrFWo/Member-Onboarding-Process?type=design&node-id=2951-6545&mode=design&t=Zaqhvd8jQUE0bVep-4)

- Move the digest opt-in to the bottom of the tags onboarding step, ensuring it remains visible over any tags displayed on the step
- Clicking anywhere on opt-in should activate the checkbox.
- Update the styling to match the designs

## Related Tickets & Documents

- Closes #19986

## QA Instructions, Screenshots, Recordings

**Loom Video**
https://www.loom.com/share/dfdcfa5f4d4846208ee0f927d11ff5b5?sid=485a361a-a4c0-4fa7-959c-25083b5ac205

**How to test?**
1. Visit `/onboarding` to test it.

**Output**

Screenshot 2023-09-19 at 3 06 26 AM
Screenshot 2023-09-19 at 3 06 41 AM

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [x] Semantic HTML implemented?
- [x] Keyboard operability supported?
- [x] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [x] Color contrast tested?

_For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

- [x] Yes

### CODE REVIEW

File: app/assets/stylesheets/preact/onboarding-modal.scss [hunk 1]
```diff
@@ -92,6 +92,36 @@
   }
 }
 
+.onboarding-content-separator {
```
Comment (rt4914) [inline app/assets/stylesheets/preact/onboarding-modal.scss:95]: Note for reviewers:

I have added these CSS classes, as I was struggling a bit to add the classes (crayons/utility) directly to html a bit.
If you have any suggestions to optimise this feel free to add. I was mainly getting stuck on background/background-color.

Other than that I have added all possible utility css classes directly in html code.

File: app/javascript/onboarding/components/FollowTags.jsx [hunk 2]
```diff
@@ -74,10 +101,31 @@ export class FollowTags extends Component {
...
+              'X-CSRF-Token': csrfToken,
+              'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({
+              notifications: {
+                email_digest_periodic: this.state.email_digest_periodic,
+              },
+            }),
+            credentials: 'same-origin',
+          });
```
Comment (jaw6) [inline app/javascript/onboarding/components/FollowTags.jsx:119]: So the form does a PATCH to `/onboarding/notifications` and a separatae POST to `/follows`, then moves to the next slide. I like this approach :+1:

**Review by jaw6 — APPROVED** [review]
Works well :+1:

**Comment by anujbhavsar96** [comment]
@rt4914 this looks great overall. Just that faded effect looks a bit harsh. I made a bit of change in [that](https://www.figma.com/file/u7VcNG4pjbn7K01V8JrFWo/Member-Onboarding-Process?type=design&node-id=2951-6338&mode=dev), can we have something like that?

File: app/javascript/onboarding/components/__tests__/FollowTags.test.jsx [hunk 3]
```diff
@@ -162,4 +162,59 @@ describe('FollowTags', () => {
       expect(getByText('0 tags selected')).toBeInTheDocument(),
     );
   });
+
```
Comment (lightalloy) [inline app/javascript/onboarding/components/__tests__/FollowTags.test.jsx:165]: Shouldn't we also test that the corresponding `fetch` should happen?

**Review by lightalloy — COMMENTED** [review]
Not very confident with reviewing frontend code, but works well. Left a small comment related to tests.

**Comment by rt4914** [comment]
@anujbhavsar96 These are the final gradient designs.

Screenshot 2023-09-20 at 4 41 36 PM
Screenshot 2023-09-20 at 4 41 47 PM

**Review by anujbhavsar96 — APPROVED** [review]
Looks great. Thanks for putting efforts into detailing ✨

File: app/javascript/onboarding/components/__tests__/FollowTags.test.jsx (hunk 3 above)
Comment (rt4914) [reply app/javascript/onboarding/components/__tests__/FollowTags.test.jsx:165]: Thanks @lightalloy I have added one last test to cover the API call too.

**Pull request merged**